pkg/querier/frontend/results_cache.go

```diff
@@ -86,6 +91,18 @@ func (s resultsCache) Do(ctx context.Context, r *queryRangeRequest) (*apiRespons
 	return response, err
 }
 
+func (s resultsCache) handleFresh(ctx context.Context, r *queryRangeRequest) (*apiResponse, []extent, error) {
+	span, ctx := opentracing.StartSpanFromContext(ctx, "handle_fresh")
+	defer span.Finish()
+
+	response, err := s.next.Do(ctx, r)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return response, nil, nil
+}
```

Comment:
I think this whole function could be reduced to `return s.next.Do(ctx, r)` - in which case, we should just inline it in `resultsCache.Do` and use the return to un-indent the following block.